File: infrastructures/ddd/typescript/src/api/dto/v1/infra/BaseResponseDto.ts
/**
 * Base class for all response DTOs.
 * Response DTOs:
 * - Live in /api/dto/v1/responses/
 * - Suffix with "Response"
 * - Used for HTTP response payloads
 * - Transport only - NO business logic
 * 
 * All response DTOs should extend this class to include common properties.
 */
export abstract class BaseResponseDto {
  public RequestId?: string;
  public Timestamp?: Date;

  constructor() {
    this.Timestamp = new Date();
  }
}

File: infrastructures/ddd/csharp/src/Api/Dto/V1/Infra/BaseResponseDto.cs
namespace OddlyDdd.Api.Dto.V1.Infra
{
    /// <summary>
    /// Base class for all response DTOs.
    /// Response DTOs are used for HTTP transport only - NO business logic.
    /// They live in /Api/Dto/ and have Request|Response|Dto suffix.
    /// </summary>
    public abstract class BaseResponseDto
    {
        // Base properties that all responses might need can go here
        // Most responses will only add specific properties in derived classes
    }
}
